src/openforms/config/tests/test_csp_update.py

```diff
@@ -27,3 +41,87 @@ def test_middleware_applies_cspsetting_models(self):
         self.assertIn("http://buzz.bazz", csp_policy["default-src"])
+
+
+class CreateCSPFormActionFromConfigTests(TestCase):
+    def setUp(self):
+        super().setUp()
+
+        self.addCleanup(clear_caches)
+
+    def test_no_ogone_nor_digid_eherkenning_config(self):
+        assert not OgoneMerchant.objects.exists()
+        assert not DigidConfiguration.objects.exists()
+        assert not EherkenningConfiguration.objects.exists()
+
+        call_command(
+            "create_csp_form_action_directives_from_config",
+            stdout=StringIO(),
+            stderr=StringIO(),
+        )
+
+        self.assertFalse(CSPSetting.objects.exists())
+
+    def test_config_records_exists_but_are_incomplete(self):
+        OgoneMerchant.objects.create(
+            endpoint_preset="",  # not realistic, but the DB allows it
+            endpoint_custom="",
+        )
+        digid_configuration = DigidConfiguration.objects.create()
+        assert not digid_configuration.idp_metadata_file
+        eherkenning_configuration = EherkenningConfiguration.objects.create()
+        assert not eherkenning_configuration.idp_metadata_file
+        CSPSetting.objects.all().delete()  # delete creates from test data setup
+
+        call_command(
+            "create_csp_form_action_directives_from_config",
+            stdout=StringIO(),
+            stderr=StringIO(),
+        )
+
+        self.assertFalse(CSPSetting.objects.exists())
+
+    def test_config_created_when_ogone_merchants_with_url_exist(self):
+        ogone_merchant = OgoneMerchantFactory.create()
+        CSPSetting.objects.all().delete()  # delete creates from test data setup
+
+        call_command(
+            "create_csp_form_action_directives_from_config",
+            stdout=StringIO(),
+            stderr=StringIO(),
+        )
+
+        csp_setting = CSPSetting.objects.get()
+        self.assertEqual(csp_setting.content_object, ogone_merchant)
+
+    def test_digid_eherkenning_config_creates_cspsetting_records(self):
+        digid_metadata = ContentFile(
+            DIGID_METADATA_POST.read_bytes(), name="digid_metadata.xml"
+        )
+        digid_configuration = DigidConfiguration.objects.create(
+            idp_metadata_file=digid_metadata
+        )
+        assert digid_configuration.idp_metadata_file
+        eherkenning_metadata = ContentFile(
+            EHERKENNING_METADATA_POST.read_bytes(), name="eh_metadata.xml"
+        )
+        eherkenning_configuration = EherkenningConfiguration.objects.create(
+            idp_metadata_file=eherkenning_metadata
+        )
+        assert eherkenning_configuration.idp_metadata_file
+        CSPSetting.objects.all().delete()  # delete creates from test data setup
+
+        call_command(
+            "create_csp_form_action_directives_from_config",
+            stdout=StringIO(),
+            stderr=StringIO(),
+        )
+
+        csp_settings = CSPSetting.objects.all()
+        self.assertEqual(len(csp_settings), 2)
```

Comment:
Is it better to use `len` instead of `CSPSetting.objects.count()`

Reply:
it depends - if you know you're going to evaluate the queryset anyway, then `len` is more efficient because a separate `.count()` performs an additional query

this version always does a single query, using `.count()` would do two queries.